### PR TITLE
New energy spear and eblades crafting

### PR DESCRIPTION
**Pull request by ThePainkiller**
## About The Pull Request
Introducing the energy spear: Craftable with a metal rod, a broken energy cutlass, some tape, a multitool, small atomcell, some cable coils, and finally trimming the cables with wirecutting  you get a bulky energy three-pronged cross spear. Sprites by Lando_o'_Rando#4602, in-hands by yours truly

image

### Reworks how to craft energy weapons by recycling broken cutlasses as a base, as follows:

Energy sword now crafted by using a broken energy cutlass , 2 steel, a welding tool, a small atomcell, cable coil and tape

Energy dagger now crafted by using a broken energy cutlass as base, 1 sheet of metal, a sawing tool, 3 sheets of plastic, a hammering tool, a small atomcell, cable coil and tape

The reason they no longer require an energy gun, the ergonomic grip, nor the subspace crystal (Star Wars reference) is because you're repairing a broken up cutlass (which is a drop from void wolves, that means to get one you **MUST** kill one of those), and since it's made up of **A LOT** of materials code-wise, you're losing out on gold and silver (plus almost a **WHOLE** stack of steel! 100 steel sheets on those!) for the sake of repurposing an e-sword, so that it balances itself out by virtue of it. It also makes the whole crafting more accessible instead of requiring very niche and specific pieces (like a spider rose), and lowers the cell size of the e-sword since the item in question is too tiny (the pommel fits on pockets) to justify a medium-sized cell (it still requires a _small_ atomcell as per _"infinite power"_ purposes). In the case of the dagger, the extra materials required are the equivalent of an ergonomic grip.

<hr>

## Changelog
:cl:
add: Added Energy Spear. Craftable with a metal rod, a broken energy cutlass, some tape, a multitool, small atomcell, some cable coils, and a wirecutting tool. Bulky, same stats as an esword, can be thrown. Icons by @Lando_o'_Rando#4602
balance: Reworked e-sword and dagger crafting. Sword now requires a broken energy cutlass, 2 steel, a welding tool, a small atomcell, cable coil and tape. Dagger now requires a broken energy cutlass, 1 metal, a sawing tool, 3 plastic, a hammering tool, cable coil and tape.
/:cl:

### CODE REVIEW

**Comment by TheSleepyCatmom** [comment]
Shouldn't be throwable, because it should function like other energy weapons and turn off when it leaves the hand.

**Comment by Dimasw99** [comment]
Deactivate upon dropped or else you have psions throwing around a 40 dmg spear

File: code/datums/craft/recipes/weapon.dm [hunk 1]
```diff
@@ -186,14 +186,30 @@
...
 		list(QUALITY_ADHESIVE, 15, 70)
 	)
 
+/datum/craft_recipe/weapon/energy_spear
+	name = "energy spear"
+	result = /obj/item/melee/energy/spear
+	icon_state = "gun"
+	steps = list(
+		list(/obj/item/stack/rods, 1, "time" = 15),
+		list(/obj/item/trash/material/e_sword_cutlass, 1, "time" = 5), // Let's tape it to a rod and call it a spear!
```
Comment (Trilbyspaceclone) [inline code/datums/craft/recipes/weapon.dm:205]: just checking if you meant the spear to use a trash esword base

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -625,7 +625,8 @@ BLIND     // can't see anything
 		/obj/item/device/radio,
 		/obj/item/clothing/mask,
 		/obj/item/storage/backpack/duffelbag/guncase,
-		/obj/item/implant/carrion_spider/holographic)
+		/obj/item/implant/carrion_spider/holographic,
+		/obj/item/melee/energy/spear)
```
Comment (Trilbyspaceclone) [inline code/modules/clothing/clothing.dm:629]: This is not needed as it already has obj.item.melee

Comment (ThePainkiller) [reply code/modules/clothing/clothing.dm:629]: Thank you catto, I must've been blind

File: code/datums/craft/recipes/weapon.dm (hunk 1 above)
Comment (ThePainkiller) [reply code/datums/craft/recipes/weapon.dm:205]: Nope, the "base" is a rod, then you line up the trashed up esword to its tip and tape it, then repair it, the rod is meant to be the base, not the trash sword!

**Comment by ThePainkiller** [comment]
> Shouldn't be throwable, because it should function like other energy weapons and turn off when it leaves the hand.

AFAIR on localtesting this PR, eswords and cutlasses (haven't tried edaggers) can be thrown without them turning themselves off, that's something tg has for sure, but I didn't see any code snippet making them turn off if they're no longer in your hands

> Deactivate upon dropped or else you have psions throwing around a 40 dmg spear

My fucking god you're right, **I HATE PSIONS**
I'm gonna make it so that it deactivates on throw, much to my chagrin since spears have always been intended to be thrown but for balance purposes (aka the existance of fucking psions) that's gonna have to go, thank you for pointing it out

**Comment by Dimasw99** [comment]
> My fucking god you're right, **I HATE PSIONS** I'm gonna make it so that it deactivates on throw, much to my chagrin since spears have always been intended to be thrown but for balance purposes (aka the existance of fucking psions) that's gonna have to go, thank you for pointing it out

Give it a short timer before deactivating so that when you throw it, it does not deactivate before it hits the target. 1 to 2 seconds should be enough for the throw duration, but not enough time for a psion to TK grab it and reuse it. (Also cope, this post was made by psion gang)

**Review by Trilbyspaceclone — APPROVED** [review]

**Review by TheSleepyCatmom — APPROVED** [review]

**Pull request merged**